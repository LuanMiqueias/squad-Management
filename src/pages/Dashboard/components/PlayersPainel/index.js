***REMOVED***
import styles from "./style.module.css";
import PlayerProfile from "../PlayerProfile/index";
const PlayersPainel = ({ data }) => {
  function orderBy(a, b) {
    console.log(a, b);
    if (a.percent > b.percent) {
      return -1;
***REMOVED***
    if (a.percent < b.percent) {
      return 1;
***REMOVED***
    return 0;
***REMOVED***
  function sortPercent(array) {
    return array.sort(orderBy);
***REMOVED***
***REMOVED***
    <div className={styles.painel}>
      {sortPercent(data)?.map((item, index) => {
      ***REMOVED***
          <PlayerProfile
            percent={item.percent}
            isMost={index === 0 ? true : false}
            title={`${index === 0 ? "Most" : "Less"} picked player`}
            name={item.name}
          ></PlayerProfile>
***REMOVED***
***REMOVED***)}
      <span className={styles.divider}></span>
    </div>
***REMOVED***
***REMOVED***

export default PlayersPainel;
