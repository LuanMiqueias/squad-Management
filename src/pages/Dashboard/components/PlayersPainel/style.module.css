.painel{
  width: 80%;
  display: flex;
  justify-content: space-between;
  padding: 32px 0;
  position: relative;
  height: 100%;
  gap:32px;
}
.divider{
  width: 1px;
  height: 100%;
  background:var(--divider);
  position: absolute;
 top: 50%;  
    left: 50%;
  transform: translate(-50%, -50%);
  display:flex;
  align-items: center;
  justify-content: center;
}
.divider::before{
  content: '';
  width: 50px;
  height: 50px;
  border-radius: 25px;
  border: 1px solid var(--divider);
  position: absolute;
}
.profile{
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  gap:16px
}
.profile h2, .profile h3{
  font-weight: 600;
  color:var(--white);
}
.player h3{
  font-size: 1rem;
  padding-right: 25px;
  border-bottom: 1px solid var(--white);
}
.player{
  display: flex;
  gap:16px;
  align-items: flex-start;
}
