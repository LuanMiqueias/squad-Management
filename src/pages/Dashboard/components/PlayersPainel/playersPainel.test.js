import { render, screen } from "@testing-library/react";
import PlayersPainel from "./";

const dataFake = [
  {
    name: "test 1",
    percent: 45,
  },
  {
    name: "test 2",
    percent: 55,
  },
];
it("Should must return in the order of the most chosen players and the least chosen players", () => {
  render(<PlayersPainel data={dataFake} />);
  const textElement = screen.getByText(/most/i);
  const textElementName_1 = screen.getByText(/t1/i);
  const textElementName_2 = screen.getByText(/t2/i);

  expect(textElement).toBeInTheDocument();
  expect(textElementName_1).toBeInTheDocument();
  expect(textElementName_2).toBeInTheDocument();
});

it("Should math snapshot", () => {
  const { container } = render(<PlayersPainel data={dataFake} />);

  expect(container.firstChild).toMatchSnapshot();
});
