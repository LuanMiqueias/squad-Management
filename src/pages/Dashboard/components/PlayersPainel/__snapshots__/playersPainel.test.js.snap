// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should math snapshot 1`] = `
<div
  class="painel"
>
  <div
    class="profile"
  >
    <h2>
      Most picked player
    </h2>
    <div
      class="player"
    >
      <div
        class="avatar borderMost"
      >
        <span>
          t2
        </span>
      </div>
      <h3>
        55
        %
      </h3>
    </div>
  </div>
  <div
    class="profile"
  >
    <h2>
      Less picked player
    </h2>
    <div
      class="player"
    >
      <div
        class="avatar borderLess"
      >
        <span>
          t1
        </span>
      </div>
      <h3>
        45
        %
      </h3>
    </div>
  </div>
  <span
    class="divider"
  />
</div>
`;
